src/NServiceBus.Persistence.DynamoDB/Outbox/OutboxPersister.cs

```diff
@@ -36,66 +37,96 @@ public OutboxPersister(IAmazonDynamoDB dynamoDbClient, OutboxPersistenceConfigur
         public async Task<OutboxMessage> Get(string messageId, ContextBag context,
             CancellationToken cancellationToken = default)
         {
-            var allItems = new List<Dictionary<string, AttributeValue>>();
+            var queryRequest = new QueryRequest
+            {
+                ConsistentRead = true,
+                KeyConditionExpression = $"#PK = :outboxId",
+                ExpressionAttributeNames =
+                    new Dictionary<string, string> { { "#PK", configuration.PartitionKeyName } },
+                ExpressionAttributeValues = new Dictionary<string, AttributeValue>
+                {
+                    { ":outboxId", new AttributeValue { S = $"OUTBOX#{endpointIdentifier}#{messageId}" } }
+                },
+                TableName = configuration.TableName
+            };
             QueryResponse response = null;
+            int numberOfTransportOperations = 0;
+            string incomingId = null;
+            List<Dictionary<string, AttributeValue>> transportOperationsAttributes = null;
             do
             {
-                var queryRequest = new QueryRequest
-                {
-                    ConsistentRead = true,
-                    KeyConditionExpression = $"#PK = :outboxId",
-                    ExclusiveStartKey = response?.LastEvaluatedKey,
-                    ExpressionAttributeNames =
-                        new Dictionary<string, string> { { "#PK", configuration.PartitionKeyName } },
-                    ExpressionAttributeValues = new Dictionary<string, AttributeValue>
-                    {
-                        { ":outboxId", new AttributeValue { S = $"OUTBOX#{endpointIdentifier}#{messageId}" } }
-                    },
-                    TableName = configuration.TableName
-                };
+                queryRequest.ExclusiveStartKey = response?.LastEvaluatedKey;
                 response = await dynamoDbClient.QueryAsync(queryRequest, cancellationToken).ConfigureAwait(false);
-                allItems.AddRange(response.Items);
+                bool headerItemSet = false;
```

Comment:
struggling with the naming, could we use some terminology like "outbox entry" and "outbox/outgoing messages"?

Reply:
Renamed